File: build-tools/codeassist-builder-plugin/src/main/java/com/tyron/builder/model/ClassField.java
package com.tyron.builder.model;

import org.jetbrains.annotations.NotNull;

import java.util.Set;

/**
 * A Simple class field with name, type and value, all as strings.
 */
public interface ClassField {
    @NotNull
    String getType();

    @NotNull
    String getName();

    @NotNull
    String getValue();

    @NotNull
    String getDocumentation();

    @NotNull
    Set<String> getAnnotations();
}

File: build-tools/codeassist-builder-plugin/src/main/java/com/tyron/builder/model/v2/dsl/ClassField.kt
package com.tyron.builder.model.v2.dsl

import com.tyron.builder.model.v2.AndroidModel

/**
 * A Simple class field with name, type and value, all as strings.
 *
 * @since 4.2
 */
interface ClassField: AndroidModel {
    val type: String
    val name: String
    val value: String
    val documentation: String
    val annotations: Set<String>
}
